magma/interface.py

```diff
@@ -159,6 +159,8 @@ def __repr__(cls):
         return str(cls)
 
     def __eq__(cls, rhs):
+        if not isinstance(rhs, InterfaceKind):
```

Comment:
This was needed to not raise an error in the single dispatch logic for interfaces